3 implement fileuploader service for image and cv storage via supabase buckets

NKSurprise opened the pull request.

iliyancherganski committed:
Created IFileUploader (not tested)

Created SupabaseBucketFileUploader : IFileUploader that can upload an image/cv(file) in the appropriate Supabase bucket. Also created  SupabaseBucketInitializer that ensures Supabase buckets exist at startup.

iliyancherganski committed:
Moved SupabaseBucketInitializer

Moved SupabaseBucketInitializer from SkillBridge.Infrastructure.Configurations to SkillBridge.Infrastructure.Supabase

NKSurprise committed:
Add user profile CRUD endpoints and service

Introduces UserProfileController with endpoints for retrieving, updating, and deleting user profiles. Adds request and response models, extends UserProfile entity with profile picture, CV upload, and GitHub connection fields, and implements IUserProfileService and UserProfileService for business logic. Updates middleware to remove unused role check.

dpS1lence committed:
Merge branch '2-create-crud-for-new-profile-pages' into dev

dpS1lence committed:
Merge pull request #6 from DEVids-VT/dev

pr

dpS1lence committed:
Refactor user profile endpoints and DI setup

Updated UserProfileController methods to remove default values from parameters and improve naming consistency. Changed UpdateUserProfileRequest.ProfilePicture to use IFormFile for file uploads. Moved service registrations for PromptBuilder, GenerateAssignmentService, and UserProfileService from Program.cs to ServiceCollectionExtensions.cs for better organization.

iliyancherganski committed:
Changed Infrastructure/Supabase to Infrasturcture/SupabaseDb

NKSurprise committed:
Add file link fields to user profile and update API

Introduces CVUpload, GitHubConnection, and ProfilePicture fields to the UserProfile entity and database schema. Updates related request/response models and controller/service logic to support these new fields, and removes unused FullName and Username properties from user profile models.

NKSurprise committed:
Update authorization policy for user profile deletion

Changed the Authorize attribute policy from 'CandidateScope' to 'Candidate' on the Delete action in UserProfileController to align with updated authorization requirements.

ImSk1 committed:
Merge branch '3-implement-fileuploader-service-for-image-and-cv-storage-via-supabase-buckets' into 2-create-crud-for-new-profile-pages

ImSk1 committed:
Merge pull request #8 from DEVids-VT/2-create-crud-for-new-profile-pages

2 create crud for new profile pages

iliyancherganski committed:
Integrate file upload and deletion in user profile service (NOT TESTED)

Refactored UserProfileService to handle file uploads and deletions for profile pictures and CVs using IFileUploader. Updated UpdateUserProfileRequest to accept IFormFile for CVUpload. Enhanced SupabaseBucketFileUploader with logging and improved error handling for file operations.

iliyancherganski committed:
Update file upload handling and request binding

Changed UpdateUserProfileRequest binding from [FromBody] to [FromForm] in UserProfileController for better form data support. Modified SupabaseBucketFileUploader to check for non-empty file names before generating URLs and adjusted return values to improve file URL handling.

iliyancherganski committed:
Improve file deletion and upload logic for user profiles (DeleteFileAsync does not work properly for images)

Refactored SupabaseBucketFileUploader to handle both public image URLs and raw CV storage keys when deleting files. Updated UserProfileService to always delete existing files before uploading new ones for CVs and profile pictures, ensuring proper file management and preventing orphaned files.

NKSurprise committed:
Add candidate role check to user profile creation

User profiles are now only created for authenticated users with the 'Candidate' role, determined via Auth0 ManagementApiClient. This prevents profiles from being created for users without the required role.

NKSurprise committed:
Merge branch '3-implement-fileuploader-service-for-image-and-cv-storage-via-supabase-buckets' of https://github.com/DEVids-VT/SkillBridge-API into 3-implement-fileuploader-service-for-image-and-cv-storage-via-supabase-buckets

NKSurprise committed:
Change IsCandidate method to private

The IsCandidate method in EnsureUserProfileMiddleware was changed from public to private to restrict its access to within the class, as it is only used internally.

iliyancherganski committed:
Refactor file deletion and update user profile logic (WORKS)

Changed SupabaseBucketFileUploader.DeleteFileAsync to accept file keys directly instead of URLs, simplifying file deletion logic. Updated UserProfileService to handle null uploads and reset fields to empty strings when uploads are not provided, improving robustness of profile updates.

ImSk1 commented on src/SkillBridge/Models/Entities/UserProfile.cs:
Do not put string.Empty. Use default!; instead.

ImSk1 commented on src/SkillBridge/Models/Response/UserProfileResponse.cs:
Do not put string.Empty. Use default!; instead.

ImSk1 commented on src/SkillBridge/Services/UserProfile/UserProfileService.cs:
```diff
@@ -0,0 +1,167 @@
+            }
+            var username = (await _managementApiClient.Users.GetAsync(userProfile.Id)).UserName;
+
+            _logger.LogInformation("Pofile found: {PofileName}", username);
```
Typo.

NKSurprise committed:
Add file upload fields and validation to user profiles

Introduces ProfilePicture, CVUpload, and GitHubConnection fields to the UserProfile entity and configures their database columns. Adds validation for file uploads and GitHub connection format. Updates controller endpoints and response models to support the new fields. Includes migration for column renaming and updates validation constants for file size and GitHub connection length.

NKSurprise committed:
Centralize file type validation and update UserProfile routes

Moved allowed file type and extension constants to ValidationConstants for reuse and maintainability. Updated SupabaseBucketFileUploader to use these centralized constants for validation. Modified UserProfileController to accept optional userId in route parameters for GET and PUT endpoints, and adjusted Update method parameter order.
